CONTRIBUTING.md

```diff
@@ -1,7 +1,180 @@
+We are always looking for new developers, so if you're new,
+please check out our [Getting Started guide](https://wiki.multitheftauto.com/wiki/Coding_info).
+
+But before we can start accepting your code, there are a couple of
+things you should know about how we work. 
+
+This document mostly contains guidelines and rules as to how your
+code should be structured and how it can be committed without
+upsetting any fellow contributors.
+
+## Where to code
+
+As a new potential contributor, you will need to fork our repository and make
+commits to your own "branch". Then you can send us a pull request.
+
+Our _`master`_ branch is the main development branch containing the
+latest, bleeding-edge code.
+
+Our _other_ branches contain groundbreaking research, radical ideas and other
+work-in-progress changes that are meant to be merged into `master` at
+a later point in time.
+
+If you're a collaborator, it's your choice whether to push branches to this
+repository or to your own fork.
+
+**Branches are "topical" and should not be "personal" to each
+user.** This means that a branch should be created for a new feature,
+not for a user specific playground.
+
+## What to code
+
+Generally, please try submit pull requests that resolve existing
+[issues](https://github.com/multitheftauto/mtasa-blue/issues).
+
+If you're looking for something to work on, take a look at the ["good first issue"]
+label, or our [milestones].
+
+["good first issue"]: https://github.com/multitheftauto/mtasa-blue/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22
+[milestones]: https://github.com/multitheftauto/mtasa-blue/milestones?direction=asc&sort=due_date
+
+Of course, if you're interested in something else, feel free to experiment
+and submit it. But discussing the feature beforehand, in an issue, will
+make your pull request more likely to be merged in a timely fashion.
+
+## Committing code
+
+**Make sure your code contributions follow the [Style Guide]**.
+
+[Style Guide]: https://github.com/multitheftauto/mtasa-blue/wiki/Style-Guide
+
+**Commits should be tested when added to master.** Commits
+that 'need to be fixed later' which directly affect the state of
+the mod will be reverted other than in exceptional circumstances.
+
+**Commit messages should**
+
+- be consistent
+- always give a clear indication of what has been changed without having to look at the code
+- include issue numbers, using [GitHub keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) where necessary
+- [follow the seven rules identified here](http://chris.beams.io/posts/git-commit/)
+    
+The most important of the [seven rules](http://chris.beams.io/posts/git-commit/) has been copied below, but please read the article:
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
```

Comment:
Not sure if it's really neccessary for us to have a strict length limit. Those length rules are more commonly used for projects where people commonly use terminal based editors etc. We aren't following that rule at all right now either. I did some fancy bash scripting and out of the 100 most recent commits, 40 of them would violate that rule. Among the entire repository commit tree, 3687/7536 commits have a subject line above 50 characters.

And because I have that in a shell right now. Here's the number for more values:

For the past 2000 commits (to stay after our Git move, earlier SVN commits are much longer, 1196/7536 are above 90 chars total)
 
- 761/2000: Above 50 
- 410/2000: Above 60 
- 230/2000: Above 70 
- 130/2000: Above 80 
- 82/2000: Above 90

Reply:
I'm not rigid on this one. You're prolly right.

My only argument would be that GitHub collapses anything longer than 70. E.g. https://github.com/multitheftauto/mtasa-blue/commits/master?after=563f22bf40c6c613865e48fb4b6561e22d5b3874+0

So I'd suggest preferring 70 instead:

```suggestion
2. Limit the subject line to 70 characters (the [seven rules] say 50, but we think 70 is okay)
```

but if you prefer getting rid of it altogether, that's fine too

```suggestion
```

Reply:
Or something midway that allows for more flexibility around that border

```suggestion
2. Limit the subject line to around 60-80 characters (the [seven rules] say 50, but we think ~70 is okay)
```

Reply:
Any feedback on this @sbx320?

Reply:
I like the second one better, but either is fine